crates/pessimistic-proof/src/imported_bridge_exit.rs

```diff
@@ -54,4 +56,15 @@ impl ImportedBridgeExit {
             self.imported_local_exit_root,
         )
     }
+
+    pub fn hash(&self) -> Digest {
+        self.bridge_exit.hash()
+    }
+}
+
+pub fn commit_imported_bridge_exits<E: Borrow<ImportedBridgeExit>>(
+    iter: impl Iterator<Item = E>,
+) -> Digest {
+    let hashes = iter.map(|exit| exit.borrow().hash()).collect::<Vec<_>>();
+    keccak256_combine(hashes.iter().map(|hash| hash.as_slice()).collect::<Vec<_>>())
```

Comment:
If you apply the following modification/generalisation to the `keccack256_combine` function:

```diff
diff --git a/crates/pessimistic-proof/src/keccak.rs b/crates/pessimistic-proof/src/keccak.rs
index 5648c92..18770f5 100644
--- a/crates/pessimistic-proof/src/keccak.rs
+++ b/crates/pessimistic-proof/src/keccak.rs
@@ -38,13 +38,14 @@ pub fn keccak256(data: &[u8]) -> Digest {
 }
 
 /// Hashes the input items using a Keccak hasher with a 256-bit security level.
-pub fn keccak256_combine<'a, I>(items: I) -> Digest
+pub fn keccak256_combine<I, T>(items: I) -> Digest
 where
-    I: IntoIterator<Item = &'a [u8]>,
+    I: IntoIterator<Item = T>,
+    T: AsRef<[u8]>,
 {
     let mut hasher = Keccak::v256();
     for data in items {
-        hasher.update(data);
+        hasher.update(data.as_ref());
     }
 
     let mut output = [0u8; 32];
```

Then this can be implemented as just: 
```suggestion
    keccak256_combine(iter.map(|exit| exit.borrow().hash()))
```

That completely eliminates the intermediate `Vec`s, presumably saving a whole bunch of cycles.

Reply:
Done 👍